pkg/types/chain_reader.go

```diff
@@ -105,10 +109,10 @@ type Sequence struct {
 }
 
 type BoundContract struct {
-	Address string
-	Name    string
+	Address  string
+	Contract string
 }
 
-func (bc BoundContract) Key() string {
-	return bc.Address + "-" + bc.Name
+func (bc BoundContract) ReadKey(method string) string {
```

Comment:
Name here should match with GetLatestValue param, so either ReadName or ReadIdentifier per my suggestiin.